### PR TITLE
[Resolve #1114] Project Dependencies part 3: Resolvable iam_role

### PR DESCRIPTION
**Pull request by jfalkenstein**
This is the third in a series of pull requests that addresses #1114 , allowing Sceptre to manage its own dependencies.

## In this PR:
* iam_role becomes a resolvable property
* The temporary circular dependency that happens when a resolving an iam_role requires using the iam_role is handled gracefully.
* Integration tests and docs to clarify the resolvability of the IAM role are added.

## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [x] Commit message starts with `[Resolve #issue-number]`.
- [x] Added/Updated unit tests.
- [x] Added/Updated integration tests (if applicable).
- [x] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [x] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by jfalkenstein** [comment]
@zaro0508 @zxiiro This one is now ready for full review as an independently mergeable and deployable branch. (Note that it _does_ point at the Part 2 branch, since it builds on the changes brought in that PR).

File: docs/_source/docs/stack_config.rst [hunk 1]
```diff
@@ -235,6 +235,13 @@ CI/CD system like Jenkins.
 In order to use this argument, however, the role needs to have an AssumeRolePolicyDocument that
 permits the user to assume that role.
 
+.. warning::
+
+   If you use set the value of ``iam_role`` with ``!stack_output``, that ``iam_role``
+   will not actually be used to obtain the stack_output, but it *WILL* be used for all subsequent stack
+   actions. Therefore, it is important that the user executing the stack action have permissions to get
+   stack outputs for the stack outputting the ``iam_role``.
```
[the remark this answers is not in the document]
Comment (jfalkenstein) [reply docs/_source/docs/stack_config.rst:243]: Yeah, that's about right. Making the `iam_role` resolvable allows us to deploy a role in one stack that Sceptre will assume and use in other stacks.

File: sceptre/connection_manager.py [hunk 2]
```diff
@@ -153,11 +153,11 @@ def _get_session(self, profile, region, iam_role):
 
                 if iam_role:
                     sts_client = session.client("sts")
+                    # maximum session name length is 64 chars. 56 + "-session" = 64
+                    session_name = f'{iam_role.split("/")[-1][:56]}-session'
```
[the remark this answers is not in the document]
Comment (jfalkenstein) [reply sceptre/connection_manager.py:157]: I don't know what you mean by "what AWS does as well". We need to name the session in which we're assuming a role, but that session name can only be so long. This was previously a bug in Sceptre's code that wouldn't be caught unless you had a long session name. However, I encountered this error myself and so making sure it's truncated is the safest approach.

File: docs/_source/docs/stack_config.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (zaro0508) [reply docs/_source/docs/stack_config.rst:243]: could we add some additional docs giving users an example workflow that they could use to setup a resolvable iam_role?

File: sceptre/connection_manager.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zaro0508) [reply sceptre/connection_manager.py:157]: what i meant was that AWS is truncating long IAM role names so we have to do the same to match.  Anyways, this looks fine.

File: docs/_source/docs/stack_config.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfalkenstein) [reply docs/_source/docs/stack_config.rst:243]: Sure, yeah. I can add that.

**Comment by jfalkenstein** [comment]
@zaro0508 I added a docs page on "Sceptre and IAM". Honestly, I think it's something we've needed for a long time, but it provides the documentation you require, I think.

**Review by zaro0508 — APPROVED** [review]
thanks for the great docs @jfalkenstein

**Pull request merged**